include/circt/Dialect/FIRRTL/FIRRTLTypesImpl.td

```diff
@@ -171,31 +172,16 @@ class BaseVectorTypeImpl<string name, string ElementType, list<Trait> traits = [
     /// Get an integer ID for the field. Field IDs start at 1, and are assigned
     /// to each field in a vector in a recursive depth-first walk of all
     /// elements. A field ID of 0 is used to reference the vector itself.
-    uint64_t getFieldID(uint64_t index);
+    uint64_t getFieldID(uint64_t index) const;
 
     /// Find the element index corresponding to the desired fieldID.  If the
     /// fieldID corresponds to a field in nested under an element, it will
     /// return the index of the parent element.
-    uint64_t getIndexForFieldID(uint64_t fieldID);
+    uint64_t getIndexForFieldID(uint64_t fieldID) const;
```

Comment:
Might be neat to have interface for these to help code that dispatches on all pass-known types that have these methods, but shrug.

If nothing else it'd ensure they're all the same and save the repetition re:declaring "the same as the other aggregates".